### PR TITLE
docs: updated community contributing documentation

### PR DESCRIPTION
**Pull request by snopeks**
The TROUBLESHOOTING.md and CONTRIBUTING.md files aim to aid the community in solving common issues and contribute to the npm project in an organized way. We really appreciate our current contributors, and we'd love to see more people get involved. We hope that this is a positive step forward. Please offer feedback and suggestions!

### CODE REVIEW

File: CONTRIBUTING.md
Comment (iarna) [inline CONTRIBUTING.md]: Delete the "hierarchy of attention"…

File: CONTRIBUTING.md
Comment (iarna) [inline CONTRIBUTING.md]: Rename to maybe "npm CLI Contributor Roles and Responsibilities"

**Comment by zkat** [comment]
I'm gonna ping @watilde @kenany and @charlotteis cause your feedback would be great.

**Comment by KenanY** [comment]
LGTM!

**Comment by legodude17** [comment]
Just to clarify, when someone doesn't know the answer, they will just ignore the issue. Correct?
I feel that that way of doing things will increase the issue backlog on this repository.

**Comment by zkat** [comment]
@legodude17 yes, that's correct. As far as the issue backlog goes, I think that's acceptable if we're going to be giving correct responses. Answering whatever and hoping you hit a mark is more likely to create more load for us, because then we've both committed to engage, *and* we now have to correct the mistake. That lengthens interactions on issues significantly and we really don't have that time to spare.

**Comment by watilde** [comment]
Wow, I'm so glad to see this finally<3 Thanks for letting me know it, Kat :)

File: CONTRIBUTING.md
Comment (watilde) [inline CONTRIBUTING.md]: I have seen this expression several times. And, every time I see this, I think that there is a way to scale.
> The CLI team is small, and has many outstanding commitments to fulfil

If it is a problem that the number of committers is small, we can encourage participation and increase the number of that. That way is how some of the other popular repositories/communities are working on this problem. So as my opinion, I felt something was wrong with `team is small` as a problem to be placed for a long time.

Since npm is a platform instead of a single command and also it is a company, I understand that it is not easy to transfer the permissions. However, I personally think that it is possible to make a quick release in bug fix as semver patch level, and it is often the solution to the fundamental issues. Also, contributors could support it when you don't have enough time :)

I would appreciate if you can consider it as next step. For now, I'd like to see what will happen!

File: TROUBLESHOOTING.md
Comment (watilde) [inline TROUBLESHOOTING.md]: how about adding a topic about a permission error on here?
https://github.com/npm/npm/wiki/Troubleshooting#permission-error

File: CONTRIBUTING.md
Comment (iarna) [inline CONTRIBUTING.md]: I'm working on an all new policy document that will (hopefully) provide a path forward for onboarding new committers, as we speak!

So yes, I agree!

File: TROUBLESHOOTING.md
Comment (zkat) [inline TROUBLESHOOTING.md]: That would be lovely. The idea is that once we merge this initial draft, we can start accepting PRs to change/add sections. We're not planning on adding those sections until this draft, its format, etc, are accepted and merged.

Comment (watilde) [inline TROUBLESHOOTING.md]: Yeah, that's fine. Living documentヽ(´ー｀)ノ

**Review by watilde — APPROVED** [review]

File: CONTRIBUTING.md
Comment (Charlotteis) [inline CONTRIBUTING.md]: I envisage this document getting longer. It might be nice to do a table of contents. As a community member (someone who tries to help out in the issue tracker?) I might see myself needing to scroll down to the bottom of the page a lot :D

So a table of contents where each line links to the relevant issue (this is possible with link IDs and such). Like was done in the troubleshooting guide ✨

File: CONTRIBUTING.md
Comment (Charlotteis) [inline CONTRIBUTING.md]: I wonder if we can be a bit stronger here

"It's best to leave the issue"

Something like "please leave the issue and try another one". I'd like to get across that you _should not try_ if you don't know the answer.

File: CONTRIBUTING.md
Comment (Charlotteis) [inline CONTRIBUTING.md]: I prefer not to use the word "simple" as, it never is simple for everyone.
As we used the word "expand" in the sentence, "but for now it is a small (& important) role" seems to fit.

File: CONTRIBUTING.md
Comment (Charlotteis) [inline CONTRIBUTING.md]: All other bullet points above and below this group of points start with a capital letter, so it's good to capitalise these ones too for consistency :D

File: CONTRIBUTING.md
Comment (Charlotteis) [inline CONTRIBUTING.md]: same as above, I prefer "please leave the issue and try another one"

File: TROUBLESHOOTING.md
Comment (Charlotteis) [inline TROUBLESHOOTING.md]: remove "simply"

File: TROUBLESHOOTING.md
Comment (Charlotteis) [inline TROUBLESHOOTING.md]: remove "you may want to"

File: TROUBLESHOOTING.md
Comment (Charlotteis) [inline TROUBLESHOOTING.md]: For consistency of editing this document like other documents in the repo:

Use the `1. 2. 3. 4` bullet points you did in the contributing guide, or switch the contributing guide to use `0. 0. 0. 0.`

For consistency of reading this document:

The starting word of each bullet should be capitalised like the rest :)

File: TROUBLESHOOTING.md
Comment (Charlotteis) [inline TROUBLESHOOTING.md]: Should we mention `npm doctor` somewhere in this document as a first try?

File: TROUBLESHOOTING.md
Comment (Charlotteis) [inline TROUBLESHOOTING.md]: remove "you may want to"

File: TROUBLESHOOTING.md
Comment (Charlotteis) [inline TROUBLESHOOTING.md]: Might want to find the right guide because `idk-the-guide` wont be valid :D

File: TROUBLESHOOTING.md
Comment (Charlotteis) [inline TROUBLESHOOTING.md]: extra space after the proxy issues link can be removed

File: CONTRIBUTING.md
Comment (Charlotteis) [inline CONTRIBUTING.md]: remove "simply"

File: TROUBLESHOOTING.md
Comment (Charlotteis) [inline TROUBLESHOOTING.md]: "simply this specific problem" doesn't make sense

"Usually, strange network errors are the case of this specific problem" (if that's what you were trying to say here"

**Review by varjmes — CHANGES_REQUESTED** [review]

**Comment by Charlotteis** [comment]
🎶 I have some questions, a couple of suggestions 🎶

File: TROUBLESHOOTING.md
Comment (legodude17) [inline TROUBLESHOOTING.md]: That seems like a good idea, maybe in the first part?

**Comment by legodude17** [comment]
Just for reference, here is something I made:
https://gist.github.com/legodude17/b16c1c1fcc3683ccea9d073fb44323a2

File: TROUBLESHOOTING.md
Comment (Charlotteis) [inline TROUBLESHOOTING.md]: With a stipulation that it's only in npm `4.1.1` and above (I believe?)

Comment (legodude17) [inline TROUBLESHOOTING.md]: I thought it was `4.1.0` and above...

File: TROUBLESHOOTING.md
Comment (legodude17) [inline TROUBLESHOOTING.md]: Before this, it should make sure it is really obvious to update `npm`. WIth a link to the part farther down.

File: CONTRIBUTING.md
Comment (legodude17) [inline CONTRIBUTING.md]: @iarna That's great! I actually asked for that in #13753. 😄

File: TROUBLESHOOTING.md
Comment (zkat) [inline TROUBLESHOOTING.md]: I think this is ok, tbh. As long as it renders and doesn't look too arcane. ✨

File: TROUBLESHOOTING.md
Comment (zkat) [inline TROUBLESHOOTING.md]: A post-merge PR on this would be nice to iterate on if anyone wants to volunteer.

File: TROUBLESHOOTING.md
Comment (zkat) [inline TROUBLESHOOTING.md]: This might just be part of `npm doctor` -- see the thread https://github.com/npm/npm/pull/15756#discussion_r101497773

PRs welcome!

**Pull request merged**
